### PR TITLE
fix(thin): use volume group size to create thin pool

### PR DESCRIPTION
**Pull request by prateekpandey14**
Signed-off-by: prateekpandey14 <prateekpandey14@gmail.com>

## Pull Request template

**Why is this PR required? What issue does it fix?**:
Handle the thin pool creation which can not be greater or equal to volumegroup size.
 
**What this PR does?**:

Earlier PV size has been used while thin pool creation, which throws insufficient space error in case if thin volume
size greater or equal then the actual volumegroup size.

As part of this change we are using the half of volumegroup size to make it lesser while creating thin pool.

**Does this PR require any upgrade changes?**:
NO

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Checklist:**
- [x] Fixes #<issue number>
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [ ] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [ ] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

### CODE REVIEW

File: pkg/lvm/lvm_util.go [hunk 1]
```diff
@@ -444,3 +447,38 @@ func snapshotExists(snapVolumeName string) (bool, error) {
...
+		return ""
+	}
+	return strings.TrimSpace(string(out))
+}
+
+// getThinPoolSize gets size for a given volumegroup, compares it with
+// the requested volume size and returns the minimum size as a thin pool size
+func getThinPoolSize(vgname, volsize string) string {
+	outStr := getVGSize(vgname)
+	vg_free_size, err := strconv.Atoi(strings.TrimSpace(string(outStr)))
```
Comment (pawanpraka1) [inline pkg/lvm/lvm_util.go:466]: Atoi will return int, should we use ParseInt(s, 10, 64) here

File: pkg/lvm/lvm_util.go [hunk 2]
```diff
@@ -444,3 +447,38 @@ func snapshotExists(snapVolumeName string) (bool, error) {
...
+	if err != nil {
+		klog.Errorf("failed to convert vg_size to int, got size,:%v , %v", outStr, err)
+		return ""
+	}
+
+	vol_size, err := strconv.Atoi(strings.TrimSpace(string(volsize)))
```
Comment (pawanpraka1) [inline pkg/lvm/lvm_util.go:472]: same here.

File: pkg/lvm/lvm_util.go [hunk 3]
```diff
@@ -444,3 +447,38 @@ func snapshotExists(snapVolumeName string) (bool, error) {
...
+	if err != nil {
+		klog.Errorf("failed to convert volsize to int, got size,:%v , %v", volsize, err)
+		return ""
+	}
+
+	if vg_free_size < vol_size {
+		// reducing 1073741824 bytes (1Gi) from the total byte size to round off
+		// blocks
+		return fmt.Sprint(vg_free_size-1073741824) + "b"
```
Comment (pawanpraka1) [inline pkg/lvm/lvm_util.go:481]: 1G is needed?

File: docs/thin_provision.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
+## Thin Provisioning
```
Comment (pawanpraka1) [inline docs/thin_provision.md:1]: let us add link to this doc in Readme, "Create a Storage class" section. Letting use know if they want to create the thin volume.

File: pkg/lvm/lvm_util.go (hunk 1 above)
Comment (prateekpandey14) [reply pkg/lvm/lvm_util.go:466]: yes, changed it use ParseInt

File: docs/thin_provision.md (hunk 4 above)
Comment (prateekpandey14) [reply docs/thin_provision.md:1]: [Storageclass.md ](https://github.com/openebs/lvm-localpv/blob/master/docs/storageclasses.md) already has the thin provision parameters info, So i just add the Thin Provision feature section in main README file

**Review by pawanpraka1 — APPROVED** [review]
looks good.

**Pull request merged**